pkg/processor/build/builder.go

```diff
@@ -1552,6 +1520,72 @@ func (b *Builder) getS3FunctionItemKey() (string, error) {
 	return s3Attributes["s3ItemKey"], nil
 }
 
+func (b *Builder) parseFunctionGitCredentials() (*githttp.BasicAuth, error) {
+	cetGitCredentials := b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitCredentials"]
+
+	if cetGitCredentials != nil {
+		type GitCredentials struct {
+			Username    string `json:"username,omitempty"`
+			Password    string `json:"password,omitempty"`
+			AccessToken string `json:"accessToken,omitempty"`
+		}
+		var gitCredentials GitCredentials
+
+		if err := mapstructure.Decode(cetGitCredentials, &gitCredentials); err != nil {
+			return nil, errors.Wrap(err, "Failed to decode git credentials map")
+		}
+
+		// use access token if given
+		if gitCredentials.AccessToken != "" {
```

Comment:
what is `accessToken`? each git server platform (github / azuredevops / bitbucket) implement their own single authentication "token" and expect it to be on username / password / whatever.

to resolve all these kind of issues, and to support many kinds of servers, I highly suggest to leave these specifications to the user. let him fill in `username` and `password` according to how _his_ git server expect it to be. that way to are being more generic